src/sxg_header.c

```diff
@@ -109,7 +109,9 @@ bool sxg_header_append_integer(const char* key, uint64_t num,
   size_t nbytes =
       snprintf(integer_buffer, sizeof(integer_buffer), "%" PRIu64, num);
 
-  assert(nbytes + 1 <= sizeof(integer_buffer));
+  if (nbytes > sizeof(integer_buffer)) {
```

Comment:
Is this actually correct?  If `snprintf` writes 22 bytes it will fill up the buffer without NUL termination. As a personal choice, `snprintf` and fixed buffers rarely does what I want and I end up writing a `smprintf` wrapper which mallocs for safety reasons.

Reply:
- `nbytes` should be of type `int`.
- I also suspect that this should be `nybytes >= sizeof(integer_buffer)`.
- `nbytes` will be negative on an encoding error, and nice to check it as well.

@gaul could you elaborate how malloc will improve this?

Reply:
I have worked on several projects where someone allocates a too small buffer or messes up the return logic.  This usually happens when a future modification changes the buffer size or format string.  Here is the most recent example I recall: https://github.com/linux-on-mac/mbpfan/commit/559a2f39f10e19c38deceb4f81b746012dcf2f56

This is not to say the existing code is wrong.  `len(str(2 ** 64 - 1)) = 20` so it has room for the NUL.

Reply:
OK, I realized the point.
Length check itself is redundant, if `nbyte` becomes bigger than `sizeof(integer_buffer)`, it must be a bug of `snprintf`.
Doubting standard library implementation is not a good attitude especially in production code.

Reply:
The story of a future modification, I think it is YAGNI.
This is so small and apparent function, I don't expect that ad-hoc extension will happen.